### PR TITLE
Mslres 4940 nexus release

### PR DESCRIPTION
**Pull request by GSvensk**
*Problem* 
We need a non-snapshot version of the SDK for the system tests
Current release process to internal repository is manual and is possibly different depending on who does it
Internal release process is not documented

*Solution*
Create script which makes internal release process 
- standardized
- faster

Document what's required to make a release
It's not possible to define multiple `<repository>` under `<distributionManagement>`, introduce release profile which gives us capability to make releases to either `clx-releases` or maven.

Why not github actions?
Internal repository URL is not accessible from github runners, everything else is possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,29 @@ To skip local test
 
 in your terminal. The project will then be compiled and tested before finally being installed in your local repository.
 
+## Making an internal release
```
Comment (gkvasnsinch) [inline README.md:54]: This instruction seems a bit odd here, as this is public repo and will be visible when you open repo...maybe we could "hide" it somewhere? though I'm not sure where 😆

File: pom.xml [hunk 2]
```diff
@@ -20,10 +20,72 @@
...
+              <nexusUrl>${env.REPO_URL}</nexusUrl>
+              <skipStaging>true</skipStaging>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>prod-repository</id>
```
Comment (gkvasnsinch) [inline pom.xml:62]: internal release => public release ?

File: pom.xml [hunk 3]
```diff
@@ -20,10 +20,72 @@
...
+      </distributionManagement>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.sonatype.plugins</groupId>
+            <artifactId>nexus-staging-maven-plugin</artifactId>
+            <version>1.6.13</version>
+            <extensions>true</extensions>
+            <configuration>
+              <serverId>clx-snapshots</serverId>
```
Comment (gkvasnsinch) [inline pom.xml:52]: why do we set snapshot credentials as default one?

File: pom.xml [hunk 4]
```diff
@@ -20,10 +20,72 @@
...
+        </snapshotRepository>
+      </distributionManagement>
+    </profile>
+  </profiles>
+
   <scm>
     <url>https://github.com/sinch/sinch-java-sms</url>
-    <connection>scm:git:git://github.com/sinch/sinch-java-sms.git</connection>
-    <developerConnection>scm:git:git@github.com:sinch/sinch-java-sms.git</developerConnection>
+    <connection>scm:git:https://github.com/sinch/sinch-java-sms.git</connection>
```
Comment (gkvasnsinch) [inline pom.xml:87]: why do you change that?

File: pom.xml (hunk 2 above)
Comment (GSvensk) [reply pom.xml:62]: what do you mean?

Comment (gkvasnsinch) [reply pom.xml:62]: It is minor thing, but I've meant profiles names. Internal-release deploys to internal repo and the second one currently is  - prod-repository

File: pom.xml (hunk 4 above)
Comment (GSvensk) [reply pom.xml:87]: Laziness, I had a PAT setup but no SSH key. I'll change it back

File: release.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+set -euo
+
+INTERNAL_RELEASE=true
+
+while getopts 'ph' OPTION; do
```
Comment (gkvasnsinch) [inline release.sh:7]: Could you please describe what's happening here? As I'm not sure to be honest...

File: release.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      ;;
+  esac
+done
+shift "$(($OPTIND -1))"
+
+[ -z "$REPO_URL" ] && echo "Need to set REPO_URL" && exit 1;
+[ -z "$GPG_PASSPHRASE" ] && echo "Need to set GPG_PASSPHRASE" && exit 1;
+
+if [ "$INTERNAL_RELEASE" = true ] ; then
+  echo "Making internal release"
```
Comment (gkvasnsinch) [inline release.sh:25]: Minor: elif "Making external release"

**Review by gkvasnsinch — APPROVED** [review]

File: release.sh (hunk 5 above)
Comment (GSvensk) [reply release.sh:11]: I added comments, does it make sense now?

Comment (GSvensk) [reply release.sh:11]: Didn't see your approval :) I assume that means yes

**Review by thomasf147 — APPROVED** [review]

**Pull request merged**